src/execution_plan/ops/op_node_by_label_scan.c

```diff
@@ -97,6 +97,10 @@ static bool _ConstructIterator
 	op->L = Graph_GetLabelMatrix(op->g, op->n->label_id);
 
 	if(array_len(op->ranges) > 0) {
+		if(op->ids == NULL) {
+			op->ids   = roaring64_bitmap_create();
+			op->ID_it = roaring64_iterator_create(op->ids);
+		}
```

Comment:
Can you explain why this was added?
I can see that when `NodeByLabelScanInit ` is called we perform:
```c
if(has_ranges) {
		op->ids   = roaring64_bitmap_create();
		op->ID_it = roaring64_iterator_create(op->ids);
	}
```

And so it's unclear to me why we need this extra logic here